beets/importer.py

```diff
@@ -672,7 +673,7 @@ def manipulate_files(self, move=False, copy=False, write=False,
                     # old paths.
                     item.move(copy, link)
 
-            if write and self.apply:
+            if write and (self.apply or self.choice_flag == action.RETAG):
```

Comment:
Thanks for the clarification, and yes, indeed makes sense. However, it seems it would require a small modification to `importer.apply_choice()` in order to avoid incorrectly applying the metadata changes if it is a `RETAG` (something similar to):
```diff
@@ -1353,7 +1353,7 @@ def apply_choice(session, task):
         return
 
     # Change metadata.
-    if task.apply:
+    if task.apply and task.choice_flag == action.APPLY:
         task.apply_metadata()
         plugins.send('import_task_apply', session=session, task=task)
```
and probably a similar tweak in regards to the `align_album_level_fields` issue at https://github.com/beetbox/beets/pull/1846#discussion-diff-51828422.

Both the existing approach and your suggested one sound like a reasonable trade-off to me!